lib/WTSI/NPG/HTS/ONT/GridIONRunMonitor.pm

```diff
@@ -129,53 +148,68 @@ sub start {
+  $pm->run_on_wait(sub {
+                     $self->debug('ForkManager waiting for child process...');
+                   }, 2);
 
   my $num_errors = 0;
 
   while ($self->monitor) {
     try {
-      $self->debug('Continue ...');
-      if ($select->can_read($SELECT_TIMEOUT)) {
-        my $n = $self->inotify->poll;
-        $self->debug("$n events");
-      }
+      # Parent process
+      my @device_dirs = $self->_find_device_dirs;
+      $self->debug('Found device directories: ', pp\@device_dirs);
+
+    DEVICE: foreach my $device_dir ($self->_find_device_dirs) {
+        # It's active, so we need do nothing
+        if (exists ${$self->devices_active}{$device_dir}) {
+          my $current_pid = $self->devices_active->{$device_dir};
+          $self->debug("'$device_dir' is already being monitored by process ",
+                       "with PID $current_pid");
+          next DEVICE;
+        }
 
-      if (@{$self->device_dir_queue}) {
-        $self->debug(scalar @{$self->device_dir_queue},
-                     ' device dirs in queue');
-
-      EVENT: while (my $device_dir = shift @{$self->device_dir_queue}) {
-          # Parent process
-          my $current_pid = $in_progress{$device_dir};
-          if ($current_pid) {
-            $self->debug("$device_dir is already being monitored by process ",
-                         "with PID $current_pid");
-            next EVENT;
+        # If it's not active, this may be caused by a pause in data
+        # gathering (flow cells remain viable for days and may be
+        # re-started). If the flow cell has been active within the
+        # quiet time window, we do not restart it.
+        if (exists ${$self->devices_complete}{$device_dir}) {
+          my $now            = time;
+          my $completed_time = $self->devices_complete->{$device_dir};
+          my $elapsed        = $now - $completed_time;
+          if ($elapsed <= $self->quiet_interval) {
+            $self->debug("'$device_dir' was completed at $completed_time, ",
+                         "$elapsed seconds ago. Still in quiet interval.")
           }
+          next DEVICE;
+        }
 
-          my $log_level = Log::Log4perl->get_logger($self->meta->name)->level;
-
-          my $pid = $pm->start($device_dir) and next EVENT;
+        my $log_level = Log::Log4perl->get_logger($self->meta->name)->level;
+        my $pid = $pm->start($device_dir) and next DEVICE;
 
-          # Child process
-          my $child_pid = $PID;
-          $self->info("Started GridIONRunPublisher with PID $child_pid on ",
-                      "'$device_dir'");
-          my $logconf =
-            $self->_make_publisher_logconf($self->output_dir, $child_pid,
-                                           $log_level);
-          Log::Log4perl::init(\$logconf);
+        # Child process
+        my $child_pid = $PID;
+        $self->info("Started GridIONRunPublisher with PID $child_pid on ",
```

Comment:
This message can be printed from the parent - would be safer this way. The code could flow like this:
my $pid = $pm->start($device_dir);
if ($pid) { # parent 
    $self->info("Starting GridIONRunPublisher with PID $pid on ",
                    "'$device_dir'");
   next DEVICE; # this is not necessary now,strictly speaking
} else { #child 
  continue from line 207